src/main/java/com/faforever/api/deployment/LegacyFeaturedModDeploymentTask.java

```diff
@@ -217,7 +218,7 @@ private void updateStatus(String message) {
         .map(path -> packDirectory(path, version, targetFolder, fileIds))
         .filter(Optional::isPresent)
         .map(Optional::get)
-        .collect(Collectors.toList());
+        .collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Why use the expanded syntax here? Is it just to make it clearer?

Reply:
Sort of. Actually it is follow the API to prevent it from breaking in the future. The javadoc of .toList() says there is no guarantee on the type of the list being returned and if you need a special type of list (in our case: a mutable one) you need to specify it like this.

Reply:
Ah gotcha that makes sense. It didn't occur to me they could change the mutability of it.

Reply:
Unfortunately List.of() and similar operators do provide ImmutableList without the type system reflecting it. The collection interfaces of Java are an unfixable mess when it comes to mutability. Instead of the type system having your back, you'll just run into runtime errors if you get handed an immutable list. But for compatibility reasons there will never be an official solution.